Client/Frontend/Browser/TabManager.swift

```diff
@@ -574,6 +580,11 @@ extension TabManager {
             if let screenshotUUID = savedTab.screenshotUUID,
                let imageStore = self.imageStore {
                 tab.screenshotUUID = screenshotUUID
+                if savedTab.faviconURL != nil {
```

Comment:
Not sure if this logic needs to be inside this `if` since it doesn't rely on the screenshot. I would move it out.